signer/src/bitcoin/rpc.rs

```diff
@@ -151,17 +160,18 @@ pub struct FeeEstimate {
 }
 
 /// A client for interacting with bitcoin-core
+#[derive(Debug, Clone)]
 pub struct BitcoinCoreClient {
     /// The underlying bitcoin-core client
-    inner: bitcoincore_rpc::Client,
+    inner: Arc<bitcoincore_rpc::Client>,
```

Comment:
I'm not sure this is actually needed since it will be used via the FallbackClient which does this internally? My thinking was to keep this complexity out of the clients themselves and aggregate that all up to the fallback client.

Reply:
Oh I added this in so that I can use the plain vanilla `BitcoinCoreClient` in the `SignerContext` in my tests.

> My thinking was to keep this complexity out of the clients themselves and aggregate that all up to the fallback client.

Yeah 🤔. My implicit goal is that we can choose to not use the fallback client in tests, since then we would have fewer layers between the code that we want to test and the actual test.

Reply:
Yeah I get that goal as well, always tradeoffs :sweat_smile:. I think we should agree on a solution then since we've got two kinda right now:

1. That we keep "internal implementations" in the `*Client`s, always implement `*Interact` on the fallback client and use its `get_client()` method to get access to the inner client in tests. Then we say that thread-safety is handled by the fallback client, or
2. We always implement both the "internal implementations" and `*Interact` on the underlying clients. `*Iteract` is also implemented on the fallback client which calls-back into those implementations (within its `exec`), and the underlying client must provide thread-safety.

Reply:
I'd vote for (2). With the exception of the `BitcoinCoreClient`, I thought the clients were all thread-safe already and already implemented their `*Interact` traits. `BitcoinCoreClient` was the only one that didn't so this brings things back into alignment.

Wait, that was a little misleading. We don't have an emily client, but that should just be a `reqwest::Client` wrapper so it will be thread safe.

Reply:
Cool, let's roll with no. 2 then :) I'll make the changes over in my stacks-client PR as well